lisa/schema.py

```diff
@@ -580,6 +580,51 @@ def _generate_min_capability(self, capability: Any) -> Any:
         return min_value
 
 
+@dataclass_json()
+@dataclass()
+class NvmeOptionSettings(FeatureSettings):
+    type: str = "NVME"
+    nvme_disk_count: search_space.CountSpace = field(
+        default=search_space.IntRange(min=0),
+        metadata=metadata(decoder=search_space.decode_count_space),
+    )
+
+    def __eq__(self, o: object) -> bool:
+        assert isinstance(o, NvmeOptionSettings), f"actual: {type(o)}"
+        return self.type == o.type and self.nvme_disk_count == o.nvme_disk_count
+
+    def __repr__(self) -> str:
+        return f"nvme_disk_count:{self.nvme_disk_count}"
+
+    def __str__(self) -> str:
+        return self.__repr__()
+
+    def __hash__(self) -> int:
+        return super().__hash__()
+
+    def check(self, capability: Any) -> search_space.ResultReason:
+        result = super().check(capability)
+
+        result.merge(
+            search_space.check_countspace(
+                self.nvme_disk_count, capability.nvme_disk_count
+            ),
+            "nvme_disk_count",
+        )
+
+    def _get_key(self) -> str:
+        return f"{super()._get_key()}/{self.nvme_disk_count}"
+
+    def _generate_min_capability(self, capability: Any) -> Any:
```

Comment:
It needs an Azure overridden feature setting, because the min_capability of Azure is different. The actual disk count is decided by vm_size, it's not changeable.

Reply:
updated